### PR TITLE
Elaticsearch 6.3.2 framework with portworx support for DCOS

### PR DESCRIPTION
**Pull request by tusharrautOO7**
Initial commit.
Verified the changes on my DCOS cluster.

### CODE REVIEW

File: frameworks/elastic/universe/config.json [hunk 1]
```diff
@@ -375,7 +575,7 @@
       "type": "object",
       "additionalProperties": false,
       "properties": {
-        "xpack_enabled": {
+        "xpack_security_enabled": {
           "description": "Whether or not to enable the commercial X-Pack plugin from Elastic",
```
Comment (disrani-px) [inline frameworks/elastic/universe/config.json]: Does description need to be updated?

File: frameworks/elastic/universe-kibana/package.json [hunk 2]
```diff
@@ -2,11 +2,11 @@
   "packagingVersion": "4.0",
   "upgradesFrom": ["{{upgrades-from}}"],
```
Comment (disrani-px) [inline frameworks/elastic/universe-kibana/package.json:3]: Is upgrade from Elastic 5 supported? If not `upgradesFrom` should be empty

File: frameworks/elastic/universe/config.json (hunk 1 above)
Comment (tusharrautOO7) [reply frameworks/elastic/universe/config.json]: @disrani-px, Should I keep description as: 
"Set it to true to enable X-Pack security on all nodes. Defaults to false"

File: frameworks/elastic/src/main/dist/elasticsearch.yml [hunk 3]
```diff
@@ -11,25 +11,25 @@ network.publish_host:
   - _local_
 cluster.name: {{CLUSTER_NAME}}
 node.name: {{TASK_NAME}}
-path.data: "${MESOS_SANDBOX}/container-path/data"
-path.logs: "${MESOS_SANDBOX}/container-path/logs"
+path.data: "{MESOS_SANDBOX}/container-path/data"
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/src/main/dist/elasticsearch.yml]: Why remove the `$`? How does this work with single brace?
It needs to be either `{{}}`(mustache template) or `$`(env var) to get substituted.

File: frameworks/elastic/src/main/dist/elasticsearch.yml [hunk 4]
```diff
@@ -11,25 +11,25 @@ network.publish_host:
...
+path.logs: "{MESOS_SANDBOX}/container-path/logs"
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/src/main/dist/elasticsearch.yml]: Same here

File: frameworks/elastic/src/main/dist/elasticsearch.yml [hunk 5]
```diff
@@ -53,6 +53,10 @@ xpack.security.transport.ssl.enabled: true
 xpack.security.http.ssl.enabled: true
 xpack.security.http.ssl.client_authentication: none
 {{/SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
+{{^SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/src/main/dist/elasticsearch.yml:56]: Add this line before the security settings, that I found in the mesosphere repo:
https://github.com/mesosphere/dcos-commons/blob/elastic-2.5.0-6.3.2/frameworks/elastic/src/main/dist/elasticsearch.yml#L43
```
# https://www.elastic.co/guide/en/elasticsearch/reference/6.3/security-settings.html
```

File: frameworks/elastic/universe/package.json [hunk 6]
```diff
@@ -2,11 +2,11 @@
   "packagingVersion": "4.0",
   "upgradesFrom": ["{{upgrades-from}}"],
   "downgradesTo": ["{{downgrades-to}}"],
-  "minDcosReleaseVersion": "1.9",
+  "minDcosReleaseVersion": "1.10",
   "name": "portworx-elastic",
   "version": "{{package-version}}",
   "maintainer": "support@portworx.com",
-  "description": "Elasticsearch 5, and optionally X-Pack",
+  "description": "Elasticsearch 6.3.2, and X-Pack",
```
Comment (piyush-nimbalkar) [inline frameworks/elastic/universe/package.json]: Just keep "Elasticsearch 6, ...", else we will have to remember to modify it every release of the framework when elastic upgrades.

File: frameworks/elastic/universe/config.json (hunk 1 above)
Comment (piyush-nimbalkar) [reply frameworks/elastic/universe/config.json]: Yes keep the keys, values, descriptions, defaults similar to the original framework.

File: frameworks/elastic/universe/package.json (hunk 6 above)
Comment (tusharrautOO7) [reply frameworks/elastic/universe/package.json]: Fixed.

File: frameworks/elastic/src/main/dist/elasticsearch.yml (hunk 5 above)
Comment (tusharrautOO7) [reply frameworks/elastic/src/main/dist/elasticsearch.yml:56]: Added this line at required place.

File: frameworks/elastic/universe-kibana/package.json (hunk 2 above)
Comment (tusharrautOO7) [reply frameworks/elastic/universe-kibana/package.json:3]: Yes, Myself and Jitu has verified the upgrade from ES-5.6.5 to ES-6.3.2

File: frameworks/elastic/universe/config.json (hunk 1 above)
Comment (tusharrautOO7) [reply frameworks/elastic/universe/config.json]: Added original description.

File: frameworks/elastic/src/main/dist/elasticsearch.yml (hunk 3 above)
Comment (tusharrautOO7) [reply frameworks/elastic/src/main/dist/elasticsearch.yml]: In the original framework it is available in {{MESOS_SANDBOX}}. I will modify and validate the changes.

https://github.com/mesosphere/dcos-commons/blob/master/frameworks/elastic/src/main/dist/elasticsearch.yml#L14

File: frameworks/elastic/src/main/dist/elasticsearch.yml (hunk 4 above)
Comment (tusharrautOO7) [reply frameworks/elastic/src/main/dist/elasticsearch.yml]: In the original framework it is available in {{MESOS_SANDBOX}}. I will modify and validate the changes.

https://github.com/mesosphere/dcos-commons/blob/master/frameworks/elastic/src/main/dist/elasticsearch.yml#L15

File: frameworks/elastic/src/main/dist/elasticsearch.yml (hunk 5 above)
Comment (piyush-nimbalkar) [reply frameworks/elastic/src/main/dist/elasticsearch.yml:56]: I don't see it.

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**